File: packages/share_bridge_wechat/ios/Classes/ShareBridgeWechatPlugin.swift
import Flutter
import UIKit

public class ShareBridgeWechatPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "share_bridge_wechat", binaryMessenger: registrar.messenger())
    let instance = ShareBridgeWechatPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "initialize":
      result(nil)
    case "isInstalled":
      result(false)
    case "supports":
      result(false)
    case "shareWebPage", "shareImage":
      result([
        "code": "unsupportedContent",
        "message": "WeChat native SDK integration is not implemented yet."
      ])
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}

File: packages/share_bridge_wechat/android/src/main/kotlin/com/gongziming/share_bridge_wechat/ShareBridgeWechatPlugin.kt
package com.gongziming.share_bridge_wechat

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** ShareBridgeWechatPlugin */
class ShareBridgeWechatPlugin :
    FlutterPlugin,
    MethodCallHandler {
    // The MethodChannel that will the communication between Flutter and native Android
    //
    // This local reference serves to register the plugin with the Flutter Engine and unregister it
    // when the Flutter Engine is detached from the Activity
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "share_bridge_wechat")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "initialize" -> result.success(null)
            "isInstalled" -> result.success(false)
            "supports" -> result.success(false)
            "shareWebPage", "shareImage" -> {
                result.success(
                    mapOf(
                        "code" to "unsupportedContent",
                        "message" to "WeChat native SDK integration is not implemented yet."
                    )
                )
            }
            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
